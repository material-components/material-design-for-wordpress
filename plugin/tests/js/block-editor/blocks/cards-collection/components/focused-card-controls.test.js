/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import FocusedCardControls from '../../../../../../assets/src/block-editor/blocks/cards-collection/components/focused-card-controls';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <FocusedCardControls { ...props } /> );
};

const baseProps = {
	cardIndex: 0,
	style: 'masonry',
	numberOfCards: 1,
	onMoveLeftOrUp: jest.fn(),
	onMoveRightOrDown: jest.fn(),
	onRemove: jest.fn(),
};

describe( 'FocusedCardControls', () => {
	afterEach( () => {
		jest.clearAllMocks();
	} );

	it( 'matches snapshot', () => {
		const wrapper = setup( baseProps );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when style prop is `list`', () => {
		const props = { ...baseProps };
		props.style = 'list';
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when number of cards is more than 1 and the card index less than the number of cards', () => {
		const props = { ...baseProps };
		props.cardIndex = 2;
		props.numberOfCards = 4;
		const wrapper = setup( props );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
