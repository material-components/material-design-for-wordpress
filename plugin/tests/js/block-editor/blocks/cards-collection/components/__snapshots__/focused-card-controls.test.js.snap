// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusedCardControls matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-container-controls"
      >
        <div
          class="move-card"
        >
          <button
            class="material-design-card-move-button-left-up"
            disabled=""
            title="Move left"
          >
            <i
              class="material-icons"
            >
              arrow_left
            </i>
          </button>
          <button
            class="material-design-card-move-button-right-down"
            disabled=""
            title="Move right"
          >
            <i
              class="material-icons"
            >
              arrow_right
            </i>
          </button>
        </div>
        <span
          class="card-number-title"
        >
          Card #1
        </span>
        <div
          class="remove-card"
        >
          <button
            class="material-design-card-close-button"
            title="Remove card"
          >
            <i
              class="material-icons"
            >
              close
            </i>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container-controls"
    >
      <div
        class="move-card"
      >
        <button
          class="material-design-card-move-button-left-up"
          disabled=""
          title="Move left"
        >
          <i
            class="material-icons"
          >
            arrow_left
          </i>
        </button>
        <button
          class="material-design-card-move-button-right-down"
          disabled=""
          title="Move right"
        >
          <i
            class="material-icons"
          >
            arrow_right
          </i>
        </button>
      </div>
      <span
        class="card-number-title"
      >
        Card #1
      </span>
      <div
        class="remove-card"
      >
        <button
          class="material-design-card-close-button"
          title="Remove card"
        >
          <i
            class="material-icons"
          >
            close
          </i>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FocusedCardControls matches snapshot when number of cards is more than 1 and the card index less than the number of cards 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-container-controls"
      >
        <div
          class="move-card"
        >
          <button
            class="material-design-card-move-button-left-up"
            title="Move left"
          >
            <i
              class="material-icons"
            >
              arrow_left
            </i>
          </button>
          <button
            class="material-design-card-move-button-right-down"
            title="Move right"
          >
            <i
              class="material-icons"
            >
              arrow_right
            </i>
          </button>
        </div>
        <span
          class="card-number-title"
        >
          Card #3
        </span>
        <div
          class="remove-card"
        >
          <button
            class="material-design-card-close-button"
            title="Remove card"
          >
            <i
              class="material-icons"
            >
              close
            </i>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container-controls"
    >
      <div
        class="move-card"
      >
        <button
          class="material-design-card-move-button-left-up"
          title="Move left"
        >
          <i
            class="material-icons"
          >
            arrow_left
          </i>
        </button>
        <button
          class="material-design-card-move-button-right-down"
          title="Move right"
        >
          <i
            class="material-icons"
          >
            arrow_right
          </i>
        </button>
      </div>
      <span
        class="card-number-title"
      >
        Card #3
      </span>
      <div
        class="remove-card"
      >
        <button
          class="material-design-card-close-button"
          title="Remove card"
        >
          <i
            class="material-icons"
          >
            close
          </i>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FocusedCardControls matches snapshot when style prop is \`list\` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-container-controls"
      >
        <div
          class="move-card"
        >
          <button
            class="material-design-card-move-button-left-up"
            disabled=""
            title="Move up"
          >
            <i
              class="material-icons"
            >
              arrow_left
            </i>
          </button>
          <button
            class="material-design-card-move-button-right-down"
            disabled=""
            title="Move down"
          >
            <i
              class="material-icons"
            >
              arrow_right
            </i>
          </button>
        </div>
        <span
          class="card-number-title"
        >
          Card #1
        </span>
        <div
          class="remove-card"
        >
          <button
            class="material-design-card-close-button"
            title="Remove card"
          >
            <i
              class="material-icons"
            >
              close
            </i>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-container-controls"
    >
      <div
        class="move-card"
      >
        <button
          class="material-design-card-move-button-left-up"
          disabled=""
          title="Move up"
        >
          <i
            class="material-icons"
          >
            arrow_left
          </i>
        </button>
        <button
          class="material-design-card-move-button-right-down"
          disabled=""
          title="Move down"
        >
          <i
            class="material-icons"
          >
            arrow_right
          </i>
        </button>
      </div>
      <span
        class="card-number-title"
      >
        Card #1
      </span>
      <div
        class="remove-card"
      >
        <button
          class="material-design-card-close-button"
          title="Remove card"
        >
          <i
            class="material-icons"
          >
            close
          </i>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
